frontend/src/components/question/QuestionAnswerEditor.tsx

```diff
@@ -37,18 +49,31 @@ export default function QuestionAnswerEditor(props: { onSubmit?: (content: strin
 				{ t('dashboard.questionView.answerEditor.title') }
 			</h2>
 			
-			<TextEditor placeholder={ t('dashboard.questionView.answerEditor.placeholder') } onInput={ setContent }
-						disabled={ hasBeenSubmitted }/>
-			
-			<div style={ { marginTop: "var(--spacing)" } }/>
-			<Button buttonStyle={ "primary" } icon={ "fi fi-rr-paper-plane" }
-					onClick={ async () => {
-						setHasBeenSubmitted(true);
-						if (props.onSubmit) await props.onSubmit(content);
-						setHasBeenSubmitted(false);
-					} }>
-				Submit
-			</Button>
+			{ props.session?.identity !== undefined
+				? <>
+					<TextEditor placeholder={ t('dashboard.questionView.answerEditor.placeholder') }
+								shouldClearContent={ shouldClearContent }
+								setShouldClearContent={ setShouldClearContent }
+								onInput={ setContent }
+								disabled={ hasBeenSubmitted }/>
+					
+					<div style={ { marginTop: "var(--spacing)" } }/>
+					<Button buttonStyle={ "primary" } icon={ "fi fi-rr-paper-plane" }
+							onClick={ async () => {
+								setHasBeenSubmitted(true);
+								if (props.onSubmit) {
+									let success = await props.onSubmit(content);
+									if (success) setShouldClearContent(true);
+								}
+								setHasBeenSubmitted(false);
+							} }>
+						Submit
+					</Button>
+				</>
+				: <Section>
+					<p>As a guest you are not permitted to submit an answer.</p>
```

Comment:
Due to i18n: Extract hardcoded text into translation file, allowing for easier translation and maintenance of multilingual requirement.